### PR TITLE
3737 validate against json schema

### PR DESCRIPTION
**Pull request by aucuparia**

### CODE REVIEW

File: datarequest.py [hunk 1]
```diff
@@ -37,20 +38,26 @@
...
 
-SCHEMA_FILENAME   = "schema.json"
-UISCHEMA_FILENAME = "uischema.json"
-DR_FILENAME       = "datarequest.json"
-PR_REV_FILENAME   = "preliminary_review.json"
-DM_REV_FILENAME   = "datamanager_review.json"
-EVAL_FILENAME     = "evaluation.json"
-ASSIGN_FILENAME   = "assignment.json"
+DRCOLLECTION      = "home/datarequests-research"
+DR                = "datarequest"
```
Comment (lwesterhof) [inline datarequest.py]: Instead of defining every filename, we could define FILENAME_EXT = ".json" once and use this.

File: datarequest.py [hunk 2]
```diff
@@ -37,20 +38,26 @@
...
+PR_REV            = "preliminary_review"
+DM_REV            = "datamanager_review"
+REV               = "review"
+ASSIGN            = "assignment"
+EVAL              = "evaluation"
```
Comment (lwesterhof) [inline datarequest.py]: EVAL -> EVALUATION
REV -> REVIEW
etc.

No need to use abbreviation. I would read REV as revision

File: util/pathutil.py [hunk 3]
```diff
@@ -107,11 +107,11 @@ def test(r, space):
 
     return (namedtuple('PathInfo', 'space zone group subpath'.split())
```
Comment (lwesterhof) [inline util/pathutil.py]: This change is not needed?

File: datarequest.py [hunk 4]
```diff
@@ -273,6 +304,10 @@ def api_datarequest_submit(ctx, data, previous_request_id):
     coll_path = "/{}/{}/{}".format(user.zone(ctx), DRCOLLECTION, request_id)
     file_path = "{}/{}".format(coll_path, DR_FILENAME)
 
+    # Validate data against schema
+    if not datarequest_data_valid(ctx, data, DR):
+        return api.Error("validation_fail", "Data did not pass validation.")
```
Comment (lwesterhof) [inline datarequest.py]: Specify what data could not pass validation?

Validation of datarequest failed?

File: datarequest.py [hunk 5]
```diff
@@ -240,10 +247,14 @@ def transform_title(row):
...
 def api_datarequest_schema_get(ctx, schema_name):
+    return datarequest_schema_get(ctx, schema_name)
+
+
+def datarequest_schema_get(ctx, schema_name):
     """Get schema and uischema of a datarequest form
 
        Arguments:
-       schema_name -- Name of schema.
+       schema_name -- Name of schema
```
Comment (lwesterhof) [inline datarequest.py:253]: Sphinx docstyle: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

```suggestion
       :param schema_name: Name of JSON schema
```

**Review by lwesterhof — CHANGES_REQUESTED** [review]

File: util/pathutil.py (hunk 3 above)
Comment (aucuparia) [reply util/pathutil.py]: Code looks fine, yes. This only fixes a flake8 error. Revert?

Comment (lwesterhof) [reply util/pathutil.py]: Should not be an error, passes in development?

Comment (aucuparia) [reply util/pathutil.py]: Probably due to running older flake8/pyflakes versions. Reverted the commit and pushed a new one.

**Review by lwesterhof — APPROVED** [review]

**Pull request merged**